src/Tasks/SystemState.cs

```diff
@@ -544,12 +549,73 @@ out fileState.frameworkName
         }
 
         /// <summary>
-        /// Cached implementation of GetDirectories.
+        /// Reads in cached data from stateFiles to build an initial cache. Avoids logging warnings or errors.
         /// </summary>
-        /// <param name="path"></param>
-        /// <param name="pattern"></param>
-        /// <returns></returns>
-        private string[] GetDirectories(string path, string pattern)
+        internal static SystemState DeserializePrecomputedCaches(ITaskItem[] stateFiles, TaskLoggingHelper log, Type requiredReturnType, GetLastWriteTime getLastWriteTime, AssemblyTableInfo[] installedAssemblyTableInfo, Func<string, bool> fileExists)
+        {
+            SystemState retVal = new SystemState();
+            retVal.SetGetLastWriteTime(getLastWriteTime);
+            retVal.SetInstalledAssemblyInformation(installedAssemblyTableInfo);
+            retVal.isDirty = stateFiles.Length > 0;
+            HashSet<string> assembliesFound = new HashSet<string>();
+            fileExists ??= FileSystems.Default.FileExists;
+
+            foreach (ITaskItem stateFile in stateFiles)
+            {
+                // Verify that it's a real stateFile; log message but do not error if not
```

Comment:
I think it was a compat concern. Old state file uses serializationMethod_1, but user is using new version of MSBuild that expected seralizationMethod_2 --> we can't read the file, but we shouldn't error. It's better to have one slow build (then revert to normal speed) than to send an error and force the user to try to figure it out. This will hopefully be an edge case anyway, since a lot of the use of these caches is with new users.